com.avaloq.tools.ddk.checkcfg.ui/src/com/avaloq/tools/ddk/checkcfg/ui/quickfix/CheckCfgQuickfixProvider.java

```diff
@@ -110,16 +118,45 @@ public void apply(final EObject element, final IModificationContext context) {
   @Fix(IssueCodes.DUPLICATE_PARAMETER_CONFIGURATION)
   public void removeDuplicateParameterConfiguration(final Issue issue, final IssueResolutionAcceptor acceptor) {
     acceptor.accept(issue, Messages.CheckCfgQuickfixProvider_REMOVE_DUPLICATE_PARAM_LABEL, Messages.CheckCfgQuickfixProvider_REMOVE_DUPLICATE_PARAM_DESCN, null, new ISemanticModification() {
+      @Override
       public void apply(final EObject element, final IModificationContext context) {
         ConfiguredCheck check = EcoreUtil2.getContainerOfType(element, ConfiguredCheck.class);
         check.getParameterConfigurations().remove(element);
       }
     });
   }
 
+  /**
+   * Removes a duplicate language configuration.
+   *
+   * @param issue
+   *          the issue
+   * @param acceptor
+   *          the acceptor
+   */
+  @Fix(IssueCodes.DUPLICATE_LANGUAGE_CONFIGURATION)
+  public void removeDuplicateLanguageConfiguration(final Issue issue, final IssueResolutionAcceptor acceptor) {
+    acceptor.accept(issue, Messages.CheckCfgQuickfixProvider_REMOVE_DUPLICATE_LANG_CONFIG_LABEL, Messages.CheckCfgQuickfixProvider_REMOVE_DUPLICATE_LANG_CONFIG_DESC, null, (element, context) -> {
+      final CheckConfiguration check = EcoreUtil2.getContainerOfType(element, CheckConfiguration.class);
+      final String languageName = ((ConfiguredLanguageValidator) element).getLanguage();
+      if (check != null && languageName != null) {
+        final List<ConfiguredLanguageValidator> allMatchingLanguages = check.getLanguageValidatorConfigurations().stream().filter(langName -> languageName.equals(langName.getLanguage())).collect(Collectors.toList());
+        if (allMatchingLanguages.size() <= 1) {
+          return; // Something went wrong. Validation guarantees 2.
+        }
+        for (ConfiguredLanguageValidator duplicate : IterableExtensions.tail(allMatchingLanguages)) {
+          // For each matching language after the first, merge contents into first and remove the language
+          IterableExtensions.head(allMatchingLanguages).getParameterConfigurations().addAll(duplicate.getParameterConfigurations());

```

Comment:
You will want to apply each of the per-language properties from each block only to each of the catalogs configured in that block and remove them from the language level, otherwise you are changing the effective configuration. See use case mentioned above.